### ISSUE TITLE
Build fails with clang

### ISSUE THREAD

**Issue by davidschlachter** [jorio/Pomme#2]
On FreeBSD 13.0-BETA1, the build fails using clang built from [the latest sources](https://github.com/llvm/llvm-project) (version 13), as well as with the most recent release version (11), giving the following errors. Would this be because of [incomplete C++20 support](https://clang.llvm.org/cxx_status.html)?

````
david@t480 0 [0] ~/Synced/bugdom
08:03:27 $ cmake -S . -B build-clang13 -DCMAKE_BUILD_TYPE=Debug -DCMAKE_C_COMPILER=/usr/home/david/llvm-head/build/bin/clang-13 -DCMAKE_CXX_COMPILER=/usr/home/david/llvm-head/build/bin/clang-13 -DCMAKE_CXX_FLAGS="-std=c++20"
-- The C compiler identification is Clang 13.0.0
-- The CXX compiler identification is Clang 13.0.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/home/david/llvm-head/build/bin/clang-13 - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/home/david/llvm-head/build/bin/clang-13 - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found SDL2: /usr/local/include/SDL2  
-- Found OpenGL: /usr/local/lib/libGL.so   
-- Found SDL2: /usr/local/include/SDL2  
-- Configuring done
-- Generating done
-- Build files have been written to: /home/david/Synced/bugdom/build-clang13
david@t480 0 [0] ~/Synced/bugdom
08:03:47 $ cmake --build build-clang13
Scanning dependencies of target Pomme
[  0%] Building CXX object extern/Pomme/CMakeFiles/Pomme.dir/src/Pomme.cpp.o
[  1%] Building CXX object extern/Pomme/CMakeFiles/Pomme.dir/src/PommeDebug.cpp.o
[  1%] Building CXX object extern/Pomme/CMakeFiles/Pomme.dir/src/Files/Files.cpp.o
[  2%] Building CXX object extern/Pomme/CMakeFiles/Pomme.dir/src/Files/HostVolume.cpp.o
/home/david/Synced/bugdom/extern/Pomme/src/Files/HostVolume.cpp:73:37: error: invalid operands to binary expression ('std::__vector_base<std::filesystem::path, std::allocator<std::filesystem::path>>::value_type' (aka 'std::filesystem::path') and 'u8string' (aka 'basic_string<char8_t>'))
        fs::path path = directories[parID] / AsU8(name);
                        ~~~~~~~~~~~~~~~~~~ ^ ~~~~~~~~~~
/usr/include/c++/v1/filesystem:1174:41: note: candidate function not viable: no known conversion from 'u8string' (aka 'basic_string<char8_t>') to 'const std::filesystem::path' for 2nd argument
  friend _LIBCPP_INLINE_VISIBILITY path operator/(const path& __lhs,
                                        ^
/home/david/Synced/bugdom/extern/Pomme/src/Files/HostVolume.cpp:160:30: error: invalid operands to binary expression ('fs::path' and 'u8string' (aka 'basic_string<char8_t>'))
        fs::path naiveConcat = path / AsU8(element);
                               ~~~~ ^ ~~~~~~~~~~~~~
/usr/include/c++/v1/filesystem:1174:41: note: candidate function not viable: no known conversion from 'u8string' (aka 'basic_string<char8_t>') to 'const std::filesystem::path' for 2nd argument
  friend _LIBCPP_INLINE_VISIBILITY path operator/(const path& __lhs,
                                        ^
/home/david/Synced/bugdom/extern/Pomme/src/Files/HostVolume.cpp:186:12: error: no matching function for call to 'FromU8'
                auto f = FromU8(candidate.path().filename().u8string());
                         ^~~~~~
/home/david/Synced/bugdom/extern/Pomme/src/Utilities/StringUtils.h:15:13: note: candidate function not viable: no known conversion from 'basic_string<char, char_traits<char>, allocator<char>>' to 'const basic_string<char8_t, char_traits<char8_t>, allocator<char8_t>>' for 1st argument
std::string FromU8(const u8string);
            ^
3 errors generated.
gmake[2]: *** [extern/Pomme/CMakeFiles/Pomme.dir/build.make:121: extern/Pomme/CMakeFiles/Pomme.dir/src/Files/HostVolume.cpp.o] Error 1
gmake[1]: *** [CMakeFiles/Makefile2:191: extern/Pomme/CMakeFiles/Pomme.dir/all] Error 2
gmake: *** [Makefile:103: all] Error 2
````

**Comment by jorio** [comment]
Can you try changing lines 3 and 4 of [src/CompilerSupport/filesystem.h](https://github.com/jorio/Pomme/blob/master/src/CompilerSupport/filesystem.h#L3) to be just `#if 0`? That will force the compiler to use an alternate "filesystem" implementation rather than your system's default.

**Comment by davidschlachter** [comment]
Thank you! With the change, the build completes without any errors. I tested a quick fix in #3, working on FreeBSD 12, clang 11.
